src/main/java/cws/k8s/scheduler/memory/LinearPredictor.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2023, Florian Friederici. All rights reserved.
+ * 
+ * This code is free software: you can redistribute it and/or modify it under 
+ * the terms of the GNU General Public License as published by the Free 
+ * Software Foundation, either version 3 of the License, or (at your option) 
+ * any later version.
+ * 
+ * This code is distributed in the hope that it will be useful, but WITHOUT ANY 
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS 
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more 
+ * details.
+ * 
+ * You should have received a copy of the GNU General Public License along with 
+ * this work. If not, see <https://www.gnu.org/licenses/>. 
+ */
+
+package cws.k8s.scheduler.memory;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.math3.stat.regression.SimpleRegression;
+
+import cws.k8s.scheduler.model.Task;
+import lombok.extern.slf4j.Slf4j;
+
+//@formatter:off
+/**
+ * LinearPredictor will use the following strategy:
+ * 
+ * If there are less than 2 observations, give no prediction, else:
+ * Calculate linear regression model and provide predictions.
+ * 
+ * Predictions start with 10% over-provisioning. If tasks fail, this will
+ * increase automatically.
+ * 
+ * @author Florian Friederici
+ *
+ */
+//@formatter:on
+@Slf4j
+public class LinearPredictor implements MemoryPredictor {
+    
+    Map<String, SimpleRegression> model;
+    Map<String, Double> overprovisioning;
+    
+    public LinearPredictor() {
+        model = new HashMap<>();
+        overprovisioning = new HashMap<>();
+    }
+    
+    @Override
+    public void addObservation(Observation o) {
+        log.debug("LinearPredictor.addObservation({})", o);
+        if (!TaskScaler.checkObservationSanity(o)) {
+            log.warn("dismiss observation {}", o);
+            return;
+        }
+
+        if (!overprovisioning.containsKey(o.task)) {
+            overprovisioning.put(o.task, 1.1);
+        }
+        
+        if (Boolean.TRUE.equals(o.success)) {
+            if (!model.containsKey(o.task)) {
+                model.put(o.task, new SimpleRegression());
+            }
+            
+            double x = o.getInputSize();
+            double y = o.getPeakRss().doubleValue();
+            model.get(o.task).addData(x,y);
+        } else {
+            log.debug("overprovisioning value will increase due to task failure");
+            Double old = overprovisioning.get(o.task);
+            overprovisioning.put(o.task, old+0.05);
```

Comment:
The overprovisioning was my first attempt in solving the problem of to low estimates.
Later I learned that [errorstrategy](https://www.nextflow.io/docs/latest/process.html#errorstrategy) is set to terminate by default and [maxretries](https://www.nextflow.io/docs/latest/process.html#maxretries) is very low (I belive this is 1 by default). So this will mostly do not occur, that this raises very much.

Reply:
How about a strategy that checks the predictions for any task in the training data and determines the highest offset needed to fit all/95%/99% values? The percent value could be a user defined hyper parameter.